src/ui/components/mixins/notebook-snapshot.js

```diff
@@ -0,0 +1,114 @@
+import $ from 'zepto';
+
+function generateTaskForm(url) {
+    var taskForm = {
+        name: "Create a Notebook Entry",
+        hint: "Please select a Notebook",
+        sections: [{
+            rows: [
+                {
+                    name: 'Entry',
+                    key: 'entry',
+                    control: 'textarea',
+                    required: false,
+                    cssClass: "l-textarea-sm"
+                },
+                {
+                    name: 'Snap Preview',
+                    key:"snapPreview",
+                    control: "snap-view",
+                    cssClass: "l-textarea-sm",
+                    src: url
+                },
+                {
+                    name: 'Save in Notebook',
+                    key: 'notebook',
+                    control: 'locator',
+                    validate: validateLocation
+                }
+            ]
+        }]
+    };
+
+    var overlayModel = {
+        title: taskForm.name,
+        message: 'Notebook Snapshot',
+        structure: taskForm,
+        value: {'entry': ""}
+    };
+
+    function validateLocation(newParentObj) {
+        return newParentObj.model.type === 'notebook';
+    }
+
+    return overlayModel;
+}
+
+function saveSnapShot(dialogService, imageUrl, imageType, imageSize, embedObject) {
+    let taskForm = generateTaskForm(imageUrl);
+
+    dialogService.getDialogResponse(
+        'overlay-dialog',
+        taskForm,
+        () => taskForm.value
+    ).then(options => {
+        let snapshotObject = {
+            src: options.snapPreview || imageUrl,
+            type: imageType,
```

Comment:
imageType and imageSize are never used and should be removed to reduce complexity.

Reply:
removed